torchtune/modules/transforms/tokenizers/_hf_tokenizer.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import json
+from typing import Any, Dict, List, Optional
+
+from tokenizers import Tokenizer
+from torchtune.modules.transforms.tokenizers._utils import BaseTokenizer
+
+
+class HFTokenizer(BaseTokenizer):
+    """
+    A wrapper around Hugging Face tokenizers. This class can be used to load from a
+    Hugging Face tokenizer.json file into a torchtune BaseTokenizer.
+
+    This class will load the tokenizer.json file from tokenizer_json_path. It will
+    attempt to infer BOS and EOS token IDs from config.json if possible, and if not
+    will fallback to inferring them from generation_config.json.
+
+    Args:
+        tokenizer_json_path (str): Path to tokenizer.json file
+        config_path (Optional[str]): Path to tokenizer_config.json file. Default: None
```

Comment:
Should be called `tokenizer_config_json_path`. Yes, I know that's a mouthful, but it pattern matches with the name of the file and there's already other "configs" in a Hugging Face model dir.